tuna.go

```diff
@@ -126,8 +128,8 @@ func (c *Common) SetConnected(connected bool) {
 	c.connected = connected
 }
 
-func (c *Common) GetServerTCPConn(force bool) (net.Conn, error) {
-	err := c.CreateServerConn(force)
+func (c *Common) GetServerTCPConn(force bool, filter *IPFilter) (net.Conn, error) {
```

Comment:
No need to pass filter as a parameter as you can use `c. EntryServiceInfo.IPFilter`